src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -1228,6 +1221,11 @@ protected Engine newEngine(boolean skipTranslogRecovery, EngineConfig config) {
         return engineFactory.newReadWriteEngine(config, skipTranslogRecovery);
     }
 
+    /* create a new translog if needed. can return a null if not needed */
+    protected FsTranslog newTranslog(ShardId shardId, Settings indexSettings, IndexSettingsService indexSettingsService,
```

Comment:
Actually, now that I look at it more, you don't need the index settings if you have the `IndexSettingsService`, because you can call `indexSettingsService.getSettings()`, so we should remove the redundant argument?